Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -14,6 +14,26 @@
  * limitations under the License.
  */
 
+/* NB: proto bytes were created via:
+     echo 'TEXT_FORMAT_PROTO' \
+       | ./build/external/protobuf/src/protobuf-build/src/protoc \
+           -I./Firestore/Protos/protos \
+           -I./build/external/protobuf/src/protobuf/src/ \
+           --encode=google.firestore.v1beta1.Value \
+           google/firestore/v1beta1/document.proto \
+           > output.bin
+ * where TEXT_FORMAT_PROTO is the text format of the protobuf. (go/textformat).
```

Comment:
While not strictly prohibited, go links in open source are frowned upon because they don't mean anything outside Google's network. Try to find an equivalent in public documentation.

Reply:
I did. There isn't any. :(

Best I was able to find was this:

https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.text_format

Oh, and also this SO question asking for details:

https://stackoverflow.com/questions/18873924/what-does-the-protobuf-text-format-look-like